geetools/ee_image_collection.py

```diff
@@ -1044,12 +1047,32 @@ def reduceInterval(
         # create a reducer from user parameters
         red = getattr(ee.Reducer, reducer)() if isinstance(reducer, str) else reducer
 
+        # to avoid different behaviour between single and multi-band image we hard coded the
+        # band names to be suffixed with the reducer output names.
+        def outputNames(b):
+            return red.getOutputs().map(lambda o: ee.String(b).cat("_").cat(o))
+
+        bandNames = self._obj.first().bandNames().map(outputNames).flatten()
+
+        # original behaviour is keeping in the output the original band names which is not compatible with
+        # muli-output reducer for obvious reasons. We need to change this back to the original band names
+        # suffixed with the reducer output names to avoid non-regression issue we added the
+        # "keep_original_names" parameter that will be removed in downstream version
+        if keep_original_names is True and red.getOutputs().length().getInfo() == 1:
+            msg = (
+                "The `keep_original_names` parameter will be removed in future versions\n"
+                " and band names will not be renamed anymore. To keep the old behaviour, rename manually\n"
+                " the output like `ic = ic.map(lambda i: i.rename([<list of the original bands>]))`.\n"
+                " To drop this warning set the `keep_original_names` parameter to `False`."
+            )
+            warnings.warn(msg, category=DeprecationWarning, stacklevel=2)
+            bandNames = self._obj.first().bandNames()
+
         def reduce(ic):
             ic = ee.ImageCollection(ic)
-            timeList = ic.aggregate_array("system:time_start")
-            start, end = timeList.get(0), timeList.get(-1)
+            start = ic.aggregate_array("system:time_start").get(0)
```

Comment:
although it might imply a similar computation, I think it could be a bit more efficient to do 

``` python
start = ic.aggregate_min('system:time_start')
end = ic.aggregate_max('system:time_start')
```

Reply:
much better indeed I'll implement that once we agree on your previous comment